hal/network/ncp_client/esp32/esp32_ncp_client.cpp

```diff
@@ -169,9 +178,15 @@ int Esp32NcpClient::off() {
         return SYSTEM_ERROR_INVALID_STATE;
     }
     muxer_.stop();
+    ncpPowerState(NcpPowerState::TRANSIENT_OFF);
     espOff();
     ready_ = false;
     ncpState(NcpState::OFF);
+    parser_.destroy();
```

Comment:
To reset the `std::unique_ptr<detail::AtParserImpl> p_` to nullptr in `AtParser` so that the `Cellular.command()` won't cause any SOS if modem is powered off.